project.go

```diff
@@ -101,5 +102,11 @@ func (p *Project) Shell(appConfig *c.Dev, args []string) {
 	cmdLine := []string{p.Config.Shell, "-c",
 		fmt.Sprintf("cd %s ; %s", relativePath, strings.Join(args, " "))}
 
-	RunOnContainer(p.Name, cmdLine...)
+	err = RunOnContainer(p.Name, cmdLine...)
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			log.Infof("Command finished with exit status: %d", exitError.ExitCode())
```

Comment:
Definitly should be exiting dev with the error code of the command run... The log message might be a bit much if a command was supplied, i.e., `dev {container} sh ls -al`. If ls is unable to do its thing, it should print its error, not sure if dev should. Might make sense when the user is attempting to get a shell and fails, though would have to detect that.

Reply:
Haha hey good to hear from you. I spent the weekend reading about testing in Go to get things to pass and just saw this. There is something to be said about the log message but I'm wondering if it would help track some things down. Currently we were having a situation where unit tests would fail silently so I guess I doubled up on making that visible.